File: String/HiddenAnagram.cxx
// Hidden Anagram
// https://edabit.com/challenge/vxWGJmtjpWtu36xfj

#include <bits/stdc++.h>
using namespace std;

string lowerconcat(string s){
    string t = "";
    for(auto x : s){
        if(x==' ')
            t+="";
        else
            t+=tolower(x);
    }
    return t;
}

bool is_anagram(string a, string b){
    auto d = 0;
    if(a.size()!=b.size())
        return false;
    for (auto x(0); a[x]&&b[x]; x++){
        d+=a[x];
        d-=b[x];
    }
    return (d==0 ? true:false);
}

string hiddenAnagram(string a, string b){
    string temp = "";
    for(auto i(0); i < (int)a.length(); i++){
        for(auto j(1); j <= (int)a.length()-i; j++){
            temp = a.substr(i,j);
            if(is_anagram(temp,b))
                return temp;
        }
    }
    return "Not found";
}

int main(){
    string a = "An old west action hero actor";
    string b = "Clint Eastwood";

    string s = "Bright is the moon";
    string t = "Bongo mirth";
    
    printf("%s\n",hiddenAnagram(lowerconcat(a),lowerconcat(b)).c_str()); // noldwestactio
    printf("%s\n",hiddenAnagram(lowerconcat(s),lowerconcat(t)).c_str()); // Not found
    
    return 0;
}

File: String/HiddenAnagram.py
# Hidden Anagram
# https://edabit.com/challenge/vxWGJmtjpWtu36xfj

def lowerconcat(s:str)->str:
	t = ""
	for x in s:
		if((ord(x)>=65 and ord(x)<=90) or (ord(x)>=97 and ord(x)<=122)):
			t+=x.lower()
		else:
			t+=""
	return t

def is_anagram(a:str,b:str)->bool:
	return True if(sorted(a)==sorted(b)) else False

def hiddenAnagram(a:str,b:str)->str:
	temp = ""
	for i in range(len(a)):
		for j in range(1,len(a)+1-i):
			temp = a[i:j]
			if(is_anagram(temp,b)):
				return temp
	return "Not found"


if __name__=='__main__':
	a = "D  e b90it->?$ (c)a r...d,,#~"
	b = "bad credit"

	s = "Bright is the moon"
	t = "Bongo mirth"

	print(hiddenAnagram(lowerconcat(a),lowerconcat(b)),end="\n") # debitcard
	print(hiddenAnagram(lowerconcat(s),lowerconcat(t)),end="\n") # Not found
